### PR TITLE
Wepbpack upgrade

### PR DESCRIPTION
**Pull request by maxime-rainville**
Naive attempt at trying to upgrade webpack and the rest of the build pipeline.

Depends on https://github.com/silverstripe/webpack-config/pull/52

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -91,27 +90,33 @@
...
-    "@silverstripe/webpack-config": "^1.7.0",
-    "@storybook/addon-actions": "^3.4.10",
-    "@storybook/addon-knobs": "^3.4.10",
-    "@storybook/addon-notes": "^3.4.10",
-    "@storybook/addon-options": "^3.4.10",
-    "@storybook/addons": "^3.4.10",
-    "@storybook/react": "^3.4.10",
-    "babel-jest": "^23.6.0",
-    "babel-runtime": "^6.26.0",
+    "@silverstripe/webpack-config": "file:../webpack-config",
```
Comment (maxime-rainville) [inline package.json:98]: That bit here assumes that `webpack-config` is one folder up from `admin`.

That's just a temporary hack.

File: _config/graphql-schema-admin.yml [hunk 2]
```diff
@@ -1,7 +1,5 @@
 ---
 Name: 'graphql-schema-admin'
-Only:
```
Comment (maxime-rainville) [inline _config/graphql-schema-admin.yml:3]: I've removed bits that were GraphQLv3 specific

File: package.json [hunk 3]
```diff
@@ -1,19 +1,16 @@
...
-    "webpack": "./node_modules/webpack/webpack.js"
-  },
   "engines": {
-    "node": "^10.x"
+    "node": "^16.x"
   },
   "scripts": {
     "build": "yarn && yarn lint && yarn test && NODE_ENV=production webpack -p --bail --progress",
-    "dev": "NODE_ENV=development webpack --progress",
+    "dev": "NODE_ENV=development NODE_OPTIONS=--trace-deprecation webpack --progress",
```
Comment (maxime-rainville) [inline package.json:13]: That's some debug option to get a better idea of any deprecation warning thrown

File: package.json [hunk 4]
```diff
@@ -75,9 +74,9 @@
     "react-dom": "^16.6.1",
     "react-load-script": "^0.0.6",
     "react-redux": "^5.0.7",
-    "react-router": "^4.4.0-beta.6",
```
Comment (maxime-rainville) [inline package.json:78]: That's not actually the latest version of react-router. I didn't want to go to the trouble of rewriting all the routing API at this stage. Not sure why we a referencing the beta version here. Looks like there never was a stable version of react router 4.4 tagged.

File: package.json [hunk 5]
```diff
@@ -91,27 +90,33 @@
     "webpack-sources": "^1.1.0"
   },
   "devDependencies": {
+    "@babel/preset-env": "^7.19",
+    "@babel/preset-react": "^7.18",
+    "@babel/runtime": "^7.19",
+    "@sect/modernizr-loader": "^1.0.2",
```
Comment (maxime-rainville) [inline package.json:96]: Some of these things probably should be in webpack config, but that can be sorted out later.

File: package.json [hunk 6]
```diff
@@ -91,27 +90,33 @@
...
     "@silverstripe/eslint-config": "^0.1.0",
-    "@silverstripe/webpack-config": "^1.7.0",
-    "@storybook/addon-actions": "^3.4.10",
-    "@storybook/addon-knobs": "^3.4.10",
-    "@storybook/addon-notes": "^3.4.10",
-    "@storybook/addon-options": "^3.4.10",
-    "@storybook/addons": "^3.4.10",
-    "@storybook/react": "^3.4.10",
```
Comment (maxime-rainville) [inline package.json:101]: I removed tho story book bit for now.

File: package.json [hunk 7]
```diff
@@ -135,11 +140,9 @@
   },
   "babel": {
     "presets": [
-      "env",
-      "react"
+      "@babel/preset-env",
+      "@babel/preset-react"
```
Comment (maxime-rainville) [inline package.json:144]: Babel renamed these presets.

File: package.json [hunk 8]
```diff
@@ -135,11 +140,9 @@
...
     ],
-    "plugins": [
-      "transform-object-rest-spread"
```
Comment (maxime-rainville) [inline package.json:142]: I think that's only needed in old version of node.

File: templates/SilverStripe/Admin/Includes/BrowserWarning.ss [hunk 9]
```diff
@@ -19,16 +19,3 @@
...
-<!--[if lte IE 8]>
-<script>
-(function () {
-  var warning = document.getElementById('browser-warning-wrapper');
-  if (warning) {
-  	warning.className = warning.className + ' browser-warning-wrapper--incompatible';
-  }
-})();
-</script>
-<![endif]-->
```
Comment (maxime-rainville) [inline templates/SilverStripe/Admin/Includes/BrowserWarning.ss:34]: Don't need the IE8 warning anymore.

File: webpack.config.js [hunk 10]
```diff
@@ -37,24 +33,19 @@ const config = [
...
     plugins: [
       ...pluginJS(ENV, PATHS),
-      new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/),
-      // Most vendor libs are loaded directly into the 'vendor' bundle (through require()
-      // calls in vendor.js). This ensures that any further require() calls in other
-      // bundles aren't duplicating libs.
-      new webpack.optimize.CommonsChunkPlugin({
-        name: 'vendor',
-        minChunks: module => module.context && module.context.indexOf('/node_modules/') > -1,
+      new webpack.IgnorePlugin({ resourceRegExp: /^\.\/locale$/, contextRegExp: /moment$/ }),
```
Comment (maxime-rainville) [inline webpack.config.js:36]: That bit copies the moment local files to a folder under dist. We should aim to remove moment altogether and use native JS method.

File: webpack.config.js [hunk 11]
```diff
@@ -110,23 +101,29 @@ const config = [
     },
     output: {
       path: PATHS.DIST,
-      filename: 'styles/[name].css',
+      filename: 'styles/[name].[chunkhash].css',
+      chunkFilename: 'styles/[name].chunk.[chunkhash].css',
```
Comment (maxime-rainville) [inline webpack.config.js:105]: I'm not 100% sure what's going on here, but I think the latest version of webpack wants to split CSS file in chunks. This bits controls where the chunks go.

**Comment by GuySartorelli** [comment]
Closing in favor of https://github.com/silverstripe/silverstripe-admin/pull/1389

**Pull request closed without merging**